jenkinsfile

```diff
@@ -1,11 +1,34 @@
-stages{
-stage(Checkout') {
-checkoutRepo('adilforms', 'Nodejssample')
-}
-stage('Build'){
-}
-stage('Test'){
-}
-stage('Publish' {
-}
+pipeline {
+  agent any
+  stages {
+    stage('Checkout'){
+      steps{
+        checkout([$class: 'GitSCM',
+                  branches: [[name: env.BRANCH_NAME]],
+                 extensions: [[$class: 'CleanBeforeCheckout']],
+                               userRemoteConfigs: [[url: env.REPO_NAME]]
+                              ])
+        }
+    }
+
+    stage('Build'){
+      steps {
+        echo 'building'
+        sh 'npm install'
+      }
+    }
+
+    stage('Test'){
+      steps {
+        echo 'Testing'
+      }
+    }
+
+    stage('Publish') {
+      steps {
+        sh "echo 'publishing"
+        sh "./docker_build_push.sh"

```

Comment:
I'm not sure if we can execute a script in the workspace, pulled from github.  If not, we may have to set up a Shared Library for a script to be available to all:
https://jenkins.io/doc/book/pipeline/shared-libraries/

Or have it baked into the Jenkins node image, like we do with acreds.sh currently.